newIDE/app/src/SceneEditor/index.js

```diff
@@ -366,6 +366,32 @@ export default class SceneEditor extends Component {
     );
   };
 
+  _onObjectSelected = selectedObjectName => {
+    this.setState({
+      selectedObjectNames: [selectedObjectName],
+    });
+  };
+
+  _onAddInstanceUnderCursor = () => {
+    const objectSelected = this.state.selectedObjectNames[0];
+    const cursorPosition = this.editor.getLastCursorPosition();
+    this._addInstance(cursorPosition[0], cursorPosition[1], objectSelected);
+    this.setState({
+      selectedObjectNames: [objectSelected],
+    });
+  };
+
+  shortenedString = (inputString: string, maxlength: number) => {
+    if (!inputString) {
+      return '';
+    }
+    let resultString = inputString;
```

Comment:
I think you can shorten the whole thing by doing: 

```
export const makeEllipsis = (str: string, maxLength: number) => {
  return str.length > maxLength ? str.substring(0, maxLength) + '...' : str;
}
```

Also nitpicking: maxLength should be in camelCase, with a capital L ;)

Reply:
Thank you @4ian done :)